app/forms/addFinancialInstitution/UkAddressFormProvider.scala

```diff
@@ -60,17 +59,17 @@ class UkAddressFormProvider @Inject() extends Mappings with RegexConstants {
         regexPostcode,
         "ukAddress.error.postcode.chars",
         postCodeAllowedChars
-      ).transform[Option[String]](
-        postCode => Option(postCode),
-        _.getOrElse(throw new IllegalStateException("postCode is empty"))
-      ),
-      "country" -> text("ukAddress.error.country.required")
-        .verifying("ukAddress.error.country.required", value => countryList.exists(_.code == value))
-        .transform[Country](
-          value => countryList.find(_.code == value).getOrElse(throw new IllegalStateException(s"Country with code [$value] not found")),
-          _.code
+      ).verifying("ukAddress.error.postcode.nonUk", postcode => !Seq("GE", "JY", "IM").exists(postcode.startsWith))
```

Comment:
shouldn't this be case insensitive lookup?

Reply:
Yes it should!